test/core/gprpp/work_serializer_test.cc

```diff
@@ -51,28 +51,37 @@ using grpc_event_engine::experimental::WaitForSingleOwner;
 
 namespace grpc_core {
 namespace {
-TEST(WorkSerializerTest, NoOp) { WorkSerializer lock(GetDefaultEventEngine()); }
+TEST(WorkSerializerTest, NoOp) {
+  auto lock = std::make_unique<WorkSerializer>(GetDefaultEventEngine());
+  lock.reset();
+  WaitForSingleOwner(GetDefaultEventEngine());
```

Comment:
What would prevent this situation from happening in production code? If the WorkSerializer is the last owner of a shared_ptr<EventEngine>, and the serializer (thus engine) are destroyed before the serializer's async Run has completed, isn't there still a risk of accessing a deleted serializer & engine? It seems like the synchronization should happen somewhere else.

Reply:
Please disregard, the object lifetimes are handled well in the inner implementations.